core_v2/sources/domains.move

```diff
@@ -377,25 +379,39 @@ module aptos_names_v2::domains {
         coin::transfer<AptosCoin>(sign, config::fund_destination_address(), price);
 
         register_name_internal(sign, subdomain_name, domain_name, registration_duration_secs, price);
+
+        // Automatically assign primary name if not exists. If exists, just assign target_addr
+        let reverse_lookup_result = get_reverse_lookup(address_of(sign));
+        if (option::is_none(&reverse_lookup_result)) {
+            // If the user has no reverse lookup set, set the user's reverse lookup.
+            set_reverse_lookup(sign, subdomain_name, domain_name);
+        } else {
+            // Automatically set the name to point to the sender's address
+            set_name_address_internal(subdomain_name, domain_name, signer::address_of(sign));
+        };
     }
 
     /// A wrapper around `register_name` as an entry function.
     /// Option<String> is not currently serializable, so we have these convenience methods
     public entry fun register_domain(
+        router_signer: &signer,
         sign: &signer,
         domain_name: String,
         registration_duration_secs: u64,
     ) acquires CollectionCapabilityV2, NameRecordV2, RegisterNameEventsV1, ReverseRecord, SetNameAddressEventsV1, SetReverseLookupEventsV1 {
+        assert!(address_of(router_signer) == @router_signer, error::permission_denied(ENOT_ROUTER));
         assert!(config::unrestricted_mint_enabled(), error::permission_denied(EVALID_SIGNATURE_REQUIRED));
         register_domain_generic(sign, domain_name, registration_duration_secs);
     }
 
     public entry fun register_domain_with_signature(
```

Comment:
should those still be `public entry fun`? should we have any public entry fun in v1 and v2?